[new product] containerd (#4706)
captn3m0/endoflife.date: products/containerd.md

---
title: containerd
category: app
iconSlug: containerd
permalink: /containerd
versionCommand: containerd --version
releasePolicyLink: https://containerd.io/releases/
activeSupportColumn: Active Support
eolColumn: Extended Support
changelogTemplate: "https://github.com/containerd/containerd/releases/tag/v__LATEST__"
releaseDateColumn: true

auto:
  methods:

>   -   git: https://github.com/containerd/containerd.git

Reviewer: ```suggestion
  -   git: https://github.com/containerd/containerd.git
  -   release_table: https://containerd.io/releases/
      ignore_empty_releases: true # ignore future releases
      selector: "table"
      fields:
        releaseCycle: "Release"
        releaseDate: "Start"
        eol: "End of Life"
```

`releaseDate` and `eol` dates can also be retrieved automatically.

>       regex: ^v(?P<major>\d*)\.(?P<minor>\d+)\.(?P<patch>\d+)$

Reviewer: ```suggestion
```

0.x releases are rarely documented on endoflife.date, so the default regex is enough.

> # Support Horizon can be found at https://github.com/containerd/containerd/blob/main/RELEASES.md#support-horizon

Reviewer: ```suggestion
# Support and EOL dates can be found on https://containerd.io/releases/#support-horizon.
```

More understandable for maintainers.

releases:
-   releaseCycle: "1.7"
    releaseDate: 2023-03-10
    support: true
    eol: false
    latest: "1.7.13"
    latestReleaseDate: 2024-02-01

-   releaseCycle: "1.6"
    releaseDate: 2022-02-16
    lts: true
    support: true
    eol: false
    latest: "1.6.28"
    latestReleaseDate: 2024-02-01

-   releaseCycle: "1.5"
    releaseDate: 2021-05-03
    support: false
    eol: 2023-02-28
    latest: "1.5.18"
    latestReleaseDate: 2023-02-16

-   releaseCycle: "1.4"
    releaseDate: 2020-08-17
    support: false
    eol: 2022-03-03
    latest: "1.4.13"
    latestReleaseDate: 2022-03-02

-   releaseCycle: "1.3"
    releaseDate: 2019-09-26
    support: false
    eol: 2021-03-04
    latest: "1.3.10"
    latestReleaseDate: 2021-03-05

-   releaseCycle: "1.2"
    releaseDate: 2018-10-24
    support: false
    eol: 2020-10-15
    latest: "1.2.13"
    latestReleaseDate: 2020-02-18

-   releaseCycle: "1.1"
    releaseDate: 2018-04-23
    support: false
    eol: 2019-10-23
    latest: "1.1.8"
    latestReleaseDate: 2019-09-27

-   releaseCycle: "1.0"
    releaseDate: 2017-12-05
    support: false
    eol: 2018-12-05
    latest: "1.0.3"

>     latestReleaseDate: 2018-04-03

Reviewer: ```suggestion
-   releaseCycle: "1.7"
    releaseDate: 2023-03-10
    support: true # releaseDate(2.0) + 6 months
    eol: false # eol(1.6)
    latest: "1.7.13"
    latestReleaseDate: 2024-01-31

-   releaseCycle: "1.6"
    releaseDate: 2022-02-15
    lts: true
    support: true # same as EOL because it's an LTS.
    eol: false # max(February 15, 2025 or releaseDate(next LTS) + 6 months
    latest: "1.6.29"
    latestReleaseDate: 2024-03-08

-   releaseCycle: "1.5"
    releaseDate: 2021-05-03
    support: false
    eol: 2023-02-28
    latest: "1.5.18"
    latestReleaseDate: 2023-02-15

-   releaseCycle: "1.4"
    releaseDate: 2020-08-17
    support: false
    eol: 2022-03-03
    latest: "1.4.13"
    latestReleaseDate: 2022-03-02

-   releaseCycle: "1.3"
    releaseDate: 2019-09-26
    support: false
    eol: 2021-03-04
    latest: "1.3.10"
    latestReleaseDate: 2021-03-04

-   releaseCycle: "1.2"
    releaseDate: 2018-10-24
    support: false
    eol: 2020-10-15
    latest: "1.2.14"
    latestReleaseDate: 2020-10-15

-   releaseCycle: "1.1"
    releaseDate: 2018-04-23
    support: false
    eol: 2019-10-23
    latest: "1.1.8"
    latestReleaseDate: 2019-09-26

-   releaseCycle: "1.0"
    releaseDate: 2017-12-04
    support: false
    eol: 2018-12-05
    latest: "1.0.3"
    latestReleaseDate: 2018-04-02
```

Reviewer: After applying our automated script to check the dates and ading a few comments for future maintainers.

---

> [containerd](https://containerd.io/) is an industry-standard container runtime with an emphasis on
> simplicity, robustness, and portability. It is available as a daemon for Linux and Windows.
> It manages the complete container lifecycle of its host system, from image transfer and storage to
> container execution and supervision to low-level storage to network attachments and beyond.

> Support horizons will be defined corresponding to a release branch, identified by `<major>.<minor>`.
> Release branches will be in one of several states:
> 
> - **Next**: The next planned release branch.
> - **Active**: The release is a stable branch which is currently supported and accepting patches.
> - **Extended**: The release branch is only accepting security patches.
> - **LTS**: The release is a long term stable branch which is currently supported and accepting patches.
> - **End of Life**: The release branch is no longer supported and no new patches will be accepted.
> 
> Releases will be supported at least one year after a minor release. This means that we will accept bug
> reports and backports to release branches until the end of life date. If no new minor release has been
> made, that release will be considered supported until 6 months after the next minor is released or one year,
> whichever is longer. Additionally, releases may have an extended security support period after the end of
> the active period to accept security backports. This timeframe will be decided by maintainers before
> the end of the active status.
> 
> Long term stable (LTS) releases will be supported for at least three years after their initial minor release.
> These branches will accept bug reports and backports until the end of life date.

Reviewer: ```suggestion
containerd follows [SemVer](https://containerd.io/releases/#releases). There is usually a new minor
release every year.

Minor non-LTS releases are actively supported for at least one year with bug and security fixes.
After that, support may be extended for an additional undefined period with security fixes only.

Some minor releases are designated as Long Term Support (LTS) releases. These are supported for at
least three years with bug and security fixes, as well as patches to support its longer term
maintainability (such as dependencies updates or even sometime feature backports to support new
Kubernetes releases). There is at least a 6-month overlap between the end of life of an LTS release
and the initial version of the next LTS release.
```

WDYT of this shorter version ? It's a little bit less precise, but I think it's more readable.
